### PR TITLE
kinder: workaround RBAC flakiness  during upgrades

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR tries to improve the current workaround for RBAC flakiness during upgrades
/assign @neolit123

### CODE REVIEW

File: kinder/pkg/cluster/manager/actions/waiter.go [hunk 1]
```diff
@@ -291,14 +291,16 @@ func staticPodHasVersion(pod, version string) func(c *status.Cluster, n *status.
 func kubeletHasRBAC(major, minor uint) func(c *status.Cluster, n *status.Node) bool {
 	return func(c *status.Cluster, n *status.Node) bool {
 		for i := 0; i < 5; i++ {
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/actions/waiter.go]: should we increase the number of retries?
i think even 30 seconds here are fine.

Comment (fabriziopandini) [reply kinder/pkg/cluster/manager/actions/waiter.go]: without any real clue on the underlying problem, I prefer to make the first try only improving the validation of the kubectl output...

BTW, this are not retries, but the number of consecutive times in a row the check should pass before assuming we are in a reliable state.

File: kinder/pkg/cluster/manager/actions/waiter.go [hunk 2]
```diff
@@ -307,7 +309,7 @@ func kubeletHasRBAC(major, minor uint) func(c *status.Cluster, n *status.Node) b
 				"-n=kube-system",
 				"-o=jsonpath='{.metadata.name}'",
 			)
-			if output1 != "" && output2 != "" {
+			if strings.Trim(output1, "'") == name1 && strings.Trim(output2, "'") == name2 {
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/actions/waiter.go]: i think we can use:
```
kubectl auth can-i get --kubeconfig /etc/kubernetes/kubelet.conf --namespace kube-system configmaps/kubelet-config-x.yy
kubectl auth can-i get --kubeconfig /etc/kubernetes/kubelet.conf --namespace kube-system configmaps/kube-proxy
```

```
$ sudo kubectl auth can-i get --kubeconfig /etc/kubernetes/kubelet.conf --namespace kube-system configmaps/kube-proxy
yes
```

File: kinder/pkg/cluster/manager/actions/waiter.go (hunk 1 above)
Comment (neolit123) [reply kinder/pkg/cluster/manager/actions/waiter.go]: i think you already explained this to me, but i again forgot.
i guess what i'm confused about is why the state can suddenly become unreliable.

once RBAC is applied it should remain there :thinking:

File: kinder/pkg/cluster/manager/actions/waiter.go (hunk 2 above)
Comment (fabriziopandini) [reply kinder/pkg/cluster/manager/actions/waiter.go]: do you know when this feature was introduced? oldest kubectl running this check should be 1.13

Comment (neolit123) [reply kinder/pkg/cluster/manager/actions/waiter.go]: kubectl 1.12 has it, but i don't know when it was added.

**Comment by fabriziopandini** [comment]
@neolit123 all changes are addressed.
Most notably I also shifted from running the check from the node being upgraded instead of always running the check from the bootstrap control-plane node

**Comment by neolit123** [comment]
thanks, let's see if this helps.
/lgtm

**Pull request merged**